proxy/http2/HPACK.cc

```diff
@@ -189,6 +189,44 @@ hpack_field_is_literal(HpackField ftype)
   return ftype == HpackField::INDEXED_LITERAL || ftype == HpackField::NOINDEX_LITERAL || ftype == HpackField::NEVERINDEX_LITERAL;
 }
 
+// Try not to use memcmp(sv, sv) and strncasecmp(sv, sv) because we don't care which value comes first on a dictionary.
+// Return immediately if the lengths of given strings don't match.
+static inline bool
+match(const char *s1, int s1_len, const char *s2, int s2_len)
```

Comment:
It seems like the caller of these functions would be better served with these functions taking std::string_view parameters.

Reply:
I found use of string_view actually costs a little in some cases. It probably depends on how it'll be optimized, but I was able to prune away time spent for string_view::data() and string_view::length() by receiving unwrapped pointers and lengths. This function is really busy and tiny optimization works, so I'd like to ensure remove any possibility of unfortunate cases.

Reply:
Sounds good. If you saw an improvement by taking char* and int, then that's good. Can you please add a comment explaining this though? Something like, `// We noticed with profiling that taking char* and int was more performant than taking std::string_view.` Otherwise someone might think to refactor by taking string_view parameters since that is what the users have.